src/date.ts

```diff
@@ -0,0 +1,151 @@
+import 'globalize/dist/globalize/date';
+import 'globalize/dist/globalize/relative-time';
+import getGlobalize, { normalizeOptions } from './util/globalize';
+
+export type DateFormatterOptions = null | {
+	/**
+	 * String value indicating a skeleton, eg. { skeleton: "GyMMMd" }.
+	 * Skeleton provides a more flexible formatting mechanism than the predefined list full, long, medium, or short represented by date, time, or datetime.
+	 * Instead, they are an open-ended list of patterns containing only date field information, and in a canonical order.
+	 */
+	skeleton?: string;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { date: "full" }.
+	 */
+	date?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { time: "full" }.
+	 */
+	time?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { datetime: "full" }
+	 */
+	datetime?: DateLength;
+}
+
+export type DateLength = 'short' | 'medium' | 'long' | 'full';
```

Comment:
Looking at the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md), exports _should_ be listed alphabetically, so I'm already in violation of this by grouping types at the top of the module. That said, this particular style suggestion is just that, so if moving `type DateLength` up before the interface that uses it will improve readability, then I won't object to doing so.